File: src/webServer/Model/Tools/Logger.cs
namespace Model.Tools;

public static class Logger
{
    private const string EXTENTION = ".txt";
    private const string FILELOCATION = "../LOG/";
    private static readonly string _FileDate = "_" + DateTime.Now.ToString("dd-MM-yyyy");

    public static void WriteLog(string msg, string fileName)
    {
        string filePath = FILELOCATION + fileName.ToUpper() + _FileDate + EXTENTION;
        using StreamWriter writer = new(filePath, true);
        writer.WriteLine($"{DateTime.Now} : {msg}");
    }
}

File: src/logicServer/src/main/java/via/sep3/logicserver/shared/Logger/Logger.java
package via.sep3.logicserver.shared.Logger;

import java.io.BufferedWriter;
import java.io.FileWriter;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

public class Logger {

    public static void writeLog(String msg, String fileName) {
        LocalDate today = LocalDate.now();
        String formattedDate = today.format(DateTimeFormatter.ofPattern("dd-MM-yyyy"));
        String filename = "Log/" + fileName.toUpperCase() + "_" + formattedDate + ".txt";

        try {
            BufferedWriter writer = new BufferedWriter(new FileWriter(filename, true));
            LocalDateTime currentDateTime = LocalDateTime.now();
            String timestamp = currentDateTime.format(DateTimeFormatter.ofPattern("dd-MM-yyyy HH:mm:ss"));

            writer.append(timestamp + " : ");
            writer.append(msg + "\n");
            writer.close();
        } catch (Exception e) {
            e.printStackTrace();
            System.out.println("Exception while writing log " + e.getMessage());
        }
    }
}
